pkg/prometheus/prometheus.go

```diff
@@ -84,14 +84,14 @@ func (p *Prometheus) ScrapeJobsMetrics(jobList ...Job) error {
 			query := renderedQuery.String()
 			renderedQuery.Reset()
 			if md.Instant {
-				docsToIndex[md.MetricName+"-start"] = append(docsToIndex[md.MetricName+"-start"], p.runInstantQuery(query, md.MetricName+"-start", jobStart, eachJob.JobConfig)...)
-				docsToIndex[md.MetricName] = append(docsToIndex[md.MetricName], p.runInstantQuery(query, md.MetricName, jobEnd, eachJob.JobConfig)...)
+				docsToIndex[md.MetricName+"-start"] = append(docsToIndex[md.MetricName+"-start"], p.runInstantQuery(query, md.MetricName+"-start", jobStart, eachJob)...)
+				docsToIndex[md.MetricName] = append(docsToIndex[md.MetricName], p.runInstantQuery(query, md.MetricName, jobEnd, eachJob)...)
```

Comment:
With this, instant metrics will be scraped using the job ending timestamp as I noted previously. Can we index an extra document in the middle of the job? i.e:

```suggestion
				docsToIndex[md.MetricName+"-start"] = append(docsToIndex[md.MetricName+"-start"], p.runInstantQuery(query, md.MetricName+"-start", jobStart, eachJob)...)
				if !eachJob.ChurnStart.IsZero() {
				    docsToIndex[md.MetricName] = append(docsToIndex[md.MetricName], p.runInstantQuery(query, md.MetricName, jobEnd, eachJob.job.ChurnEnd)...)
				} else {
			    	docsToIndex[md.MetricName] = append(docsToIndex[md.MetricName], p.runInstantQuery(query, md.MetricName, jobEnd, eachJob)...)
			 }
```

Reply:
Instant queries are already not being taken into account according to the check [here](https://github.com/kube-burner/kube-burner/blob/49ae8170f19b0e2ef5104ed655e91acd36a98145/pkg/prometheus/prometheus.go#L181). I am not sure about the intention behind this suggestion.